src/server/game/AI/CoreAI/PetAI.cpp

```diff
@@ -514,9 +514,22 @@ Unit* PetAI::SelectNextTarget(bool allowAutoSelect) const
     // To prevent aggressive pets from chain selecting targets and running off, we
     // only select a random target if certain conditions are met.
     if (allowAutoSelect)
+    {
         if (!me->GetCharmInfo()->IsReturning() || me->GetCharmInfo()->IsFollowing() || me->GetCharmInfo()->IsAtStay())
+        {
             if (Unit* nearTarget = me->ToCreature()->SelectNearestTargetInAttackDistance(MAX_AGGRO_RADIUS))
-                return nearTarget;
+            {
+                if (nearTarget->IsPlayer() && nearTarget->ToPlayer()->IsPvP() && !owner->IsPvP()) // If owner is not PvP flagged and target is PvP flagged, do not attack
+                {
+                    return nullptr; /// @todo: try for another target
```

Comment:
the if could be reversed and just let it go to the last nullptr in the function

Reply:
Would have to unreverse it to deal with the todo there eventually, no?

Reply:
Oh I missed the comment kekw

Reply:
```/// @todo``` is the keyword used in doxygen for future refactors, is it the case here?